src/util/edit.cc

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright 2017 CodiLime
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "util/edit.h"
+
+namespace veles {
+namespace util {
+
+void EditEngine::changeBytes(size_t pos, const QVector<uint64_t>& bytes, const QVector<uint64_t>& old_bytes, bool add_to_history) {
+  if (add_to_history) {
+    edit_stack_data_.push_back(old_bytes);
+    edit_stack_.push_back(QPair<size_t, size_t>(pos, bytes.size()));
+    while (edit_stack_.size() > edit_stack_limit_) {
+      edit_stack_data_.pop_front();
+      edit_stack_.pop_front();
```

Comment:
Looks slow-ish, we should use QList probably. IMO it's not too important for noticeable performance, but at least add "TODO" here.